language/slackbot/demo_bot.js

```diff
@@ -0,0 +1,208 @@
+locally if you like.
+To do this, create a file containing your Slack token (as described in
+the README), then point 'slack_token_path' to that file when you run the script:
+
+    echo my-slack-token > slack-token
+    slack_token_path=./slack-token node demo_bot.js
+
+See the README.md in this directory for more information about setup and usage.
+
+*/
+
+'use strict';
+
+const Language = require('@google-cloud/language');
+const Storage = require('@google-cloud/storage');
+
+
+var Botkit = require('botkit');
+var fs = require('fs');
+
+var controller = Botkit.slackbot({debug: false});
+
+var sqlite3 = require('sqlite3').verbose();
+// create our database if it does not already exist.
+var db = new sqlite3.Database('slackDB.db');
+
+const util = require('util');
+
+// the number of most frequent entities to retrieve from the db on request.
+const NumEnts = 20;
+// The magnitude of sentiment of a posted text above which the bot will respond.
+const SentimentThresh = 30;
+
+if (!process.env.slack_token_path) {
+  console.log('Error: Specify slack_token_path in environment');
+  process.exit(1);
+}
+
+fs.readFile(process.env.slack_token_path, (err, data) => {
+  if (err) {
+    console.log('Error: Specify token in slack_token_path file');
+    process.exit(1);
+  }
+  data = String(data);
+  data = data.replace(/\s/g, "");
+  controller.spawn({token: data}).startRTM(function(err) {
+    if (err) {
+      throw new Error(err);
+    }
+  });
+});
+
+// create the table we'll use to store entity information if it does not already
+// exist.
+db.run('CREATE TABLE if not exists entities (name text, type text, ' +
+  'salience real, wiki_url text, ts integer)');
+
+
+function analyzeSentimentOfText (text) {
+  // Instantiates a client
+  const language = Language();
+
+  // Instantiates a Document, representing the provided text
+  const document = language.document({
+    // The document text, e.g. "Hello, world!"
+    content: text
+  });
+
+  // Detects the 'sentiment' of some text using the NL API
+  return document.detectSentiment()
+    .then((results) => {
+      const sentiment = results[0];
+      console.log(`Sentiment: ${sentiment >= 0 ? 'positive' : 'negative'}.`);
+      return sentiment;
+    });
+}
+
+function analyzeEntitiesOfText (text, ts) {
+  // Instantiates a client
+  const language = Language();
+
+  // Instantiates a Document, representing the provided text
+  const document = language.document({
+    // The document text, e.g. "Hello, world!"
+    content: text
+  });
+
+  // Detects entities in the document
+  return document.detectEntities()
+    .then((results) => {
+      const entities = results[0];
+      console.log('Entities:');
+      for (let type in entities) {
+        console.log(`${type}:`, entities[type]);
+      }
+      const entityList = results[1]['entities'];
+      for (let ent of entityList) {
+        // console.log(util.inspect(ent, false, null));
+        const ename = ent['name'];
```

Comment:
Throughout, use the dot operator instead of bracket notation to access object properties.